packages/openid4vc/src/shared/utils.ts

```diff
@@ -56,8 +58,17 @@ export async function getSphereonSuppliedSignatureFromJwtIssuer(
     alg = _alg
     kid = verificationMethod.id
     did = verificationMethod.controller
+  } else if (jwtIssuer.method === 'x5c') {
+    const x509Service = agentContext.dependencyManager.resolve(X509Service)
+    await x509Service.validateCertificateChain(jwtIssuer.chain)
+
+    const leafCertificate = X509Service.getLeafCertificate(jwtIssuer.chain)
+    alg = getJwkFromKey(leafCertificate.publicKey).supportedSignatureAlgorithms[0]
```

Comment:
array can be empty. We should check if alg is defined

Reply:
@auer-martin I think you get it, haha. All the comments outside of the `crypto` folder are for you :D.